src/test/kotlin/com/atlassian/performance/tools/virtualusers/DockerJiraFormula.kt

```diff
@@ -52,7 +52,12 @@ class DockerJiraFormula(
         db: SudoSshUbuntuContainer
     ): DockerJira {
         val peerJiraPort = 8080
-        val publishedJiraPort = jira.ports.bindings[ExposedPort.tcp(peerJiraPort)]!!.single().hostPortSpec.toInt()
+        val publishedJiraPort = jira
+            .ports
+            .bindings[ExposedPort.tcp(peerJiraPort)]!!
+            .single { it.hostIp == "0.0.0.0" }
```

Comment:
Could you describe why this is proper filtering (for IPv4?)? Maybe it's obvious, but I don't know where to look for it.

Reply:
The port is exposed twice, one for IPv4 and one IPv6. We need to pick one to connect with.
The `0.0.0.0` is IPv4 syntax for "all addresses".

Reply:
> The `0.0.0.0` is IPv4 syntax for "all addresses".

Will `hostIp` always be `0.0.0.0` for IPv4 ports? I can see in the javadoc of `getHostIp()` that it can also be `null`: https://javadoc.io/static/com.github.docker-java/docker-java-api/3.2.13/com/github/dockerjava/api/model/Ports.Binding.html#getHostIp--

I'm thinking though that even if something would change with `hostIp` compared to the behavior we have now the problem would manifest itself immediately and verbosely, so I guess we shouldn't care for now.

Reply:
Exactly right, sometimes we just gotta "let it crash" (like in Erlang).